biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -265,13 +263,49 @@ public TaggedData call() throws Exception {
 	}
 
 	ProgressPlugin.Task getTask(final HttpRequest< ? > request) {
+		final String name = "Download " + request.url;
+		final int size = 100;
 		final ProgressPlugin.Task task;
-		if (progress != null) {
-			task = progress.startTask("Download " + request.url, 100);
-		} else {
+		final List<ProgressPlugin> progressPlugins = registry != null ? registry.getPlugins(ProgressPlugin.class)
+				: null;
+
+		if (progressPlugins != null && progressPlugins.size() > 1) {
+			final List<ProgressPlugin.Task> multiplexedTasks = new ArrayList<>();
+
+			for (ProgressPlugin progressPlugin : progressPlugins) {
+				multiplexedTasks.add(progressPlugin.startTask(name, size));
+			}
+
 			task = new ProgressPlugin.Task() {
+				@Override
+				public void worked(int units) {
+					for (ProgressPlugin.Task multiplexedTask : multiplexedTasks) {
```

Comment:
To be nitpicky, the loop variable should be called `task` so it is the same as the other method's loops.